packages/snaps-sdk/src/jsx/components/form/AssetSelector.ts

```diff
@@ -0,0 +1,67 @@
+import type {
+  NonEip155AssetType,
+  NonEip155ChainId,
+  NonEip155MatchingAddressesCaipAccountIdList,
+} from '../../../types';
+import { createSnapComponent } from '../../component';
+
+/**
+ * The props of the {@link AssetSelector} component.
+ *
+ * @property name - The name of the asset selector. This is used to identify the
+ * state in the form data.
+ * @property addresses - The addresses of the account to pull the assets from.
+ * Only one address is supported, but different chains can be used.
```

Comment:
Why is only one address supported right now? If that's to change then I think it makes sense to update to:

```suggestion
 * Only one address is supported currently, but different chains can be used.
```

Reply:
That is part of the specs, it's not planned to change

Reply:
So why do we have this as an array then?

Reply:
This is design to take the `AccountSelector` state value directly, the account selector returns an array of CAIP-10 IDs to represent all the available networks for an address. Same here, this list allows you to specify more than one CAIP-10 id to allow multiple networks to be used.

Reply:
I think we should make it clear that you can specify multiple CAIP-10 addresses but the reference part of the address must be the same